concurrent_test.go

```diff
@@ -146,8 +146,17 @@ func concurrentReadAndWrite(t *testing.T,
 	})
 	require.NoError(t, err)
 
+	var records historyRecords
+	// t.Failed() returns false during panicking. We need to forcibly
+	// save data on panicking.
+	// Refer to: https://github.com/golang/go/issues/49929
+	panicked := true
```

Comment:
Can you use `recover()` as it's more naturally refers to detecting panic vs strage variable `paniced` and a comment?

Reply:
Note the purpose of `recover` is to capture and recover the panicking. The purpose here is only to detect the panicking

Reply:
Yea, looks like you are right.